FHEM/00_SIGNALduino.pm

```diff
@@ -2883,85 +2886,89 @@ sub SIGNALduino_IdList($@)
+		
+	if ($wflag == 0) {			# whitelist not aktive
+		if (!defined($blacklist)) {
+			$blacklist = AttrVal($name,"blacklist_IDs","");
+		}
+		if (length($blacklist) > 0) {							# Blacklist in Hash wandeln
+			SIGNALduino_Log3 $name, 3, "$name IdList: attr blacklistIds=$blacklist";
 			%BlacklistIDs = map { $_ => 1 } split(",", $blacklist);
-			my $w = join ', ' => map "$_" => keys %BlacklistIDs;
-			SIGNALduino_Log3 $name, 4, "$name Attr blacklist $w";
+			#my $w = join ', ' => map "$_" => keys %BlacklistIDs;
+			#SIGNALduino_Log3 $name, 3, "$name IdList, Attr blacklist $w";
 			$bflag = 1;
 		}
 	}
 	
 	my $id;
 	foreach $id (keys %ProtocolListSIGNALduino)
 	{
-		next if ($id eq 'id');
-		if ($wflag == 1 && !defined($WhitelistIDs{$id}))
+		if ($wflag == 1)				# whitelist aktive
 		{
-			#SIGNALduino_Log3 $name, 3, "skip ID $id";
-			next;
+			next if (!exists($WhitelistIDs{$id}))		# Id wurde in der whitelist nicht gefunden
 		}
-		if ($bflag == 1 && defined($BlacklistIDs{$id}))
-		{
-			#SIGNALduino_Log3 $name, 4, "$name skip Blacklist ID $id";
-			next;
-		}
-		
-		if (defined($ProtocolListSIGNALduino{$id}{developId}) && substr($ProtocolListSIGNALduino{$id}{developId},0,1) eq "m") {
-			my $devid = "m$id";
-			if ($develop !~ m/$devid/) {		# skip wenn die Id nicht im Attribut development steht
-				#SIGNALduino_Log3 $name, 3, "$name: ID=$devid skipped (developId=m)";
-				push (@skippedId, $devid);
+		else {						# whitelist not aktive
+			if ($bflag == 1 && exists($BlacklistIDs{$id})) {
+				#SIGNALduino_Log3 $name, 3, "$name IdList, skip Blacklist ID $id";
+				push (@skippedBlackId, $id);
 				next;
 			}
-		}																					 
-		
-		if (defined($ProtocolListSIGNALduino{$id}{developId}) && substr($ProtocolListSIGNALduino{$id}{developId},0,1) eq "p") {
-			my $devid = "p$id";
-			if ($develop !~ m/$devid/) {		# skip wenn die Id nicht im Attribut development steht
-				#SIGNALduino_Log3 $name, 3, "$name: ID=$devid skipped (developId=p)";
-				push (@skippedId, $devid);
-				next;
-			}
-		}
 		
-		if (defined($ProtocolListSIGNALduino{$id}{developId}) && substr($ProtocolListSIGNALduino{$id}{developId},0,1) eq "y") {
-			my $devid = "y$id";
-			if ($develop !~ m/$devid/) {		# skip wenn die Id nicht im Attribut development steht
-				#SIGNALduino_Log3 $name, 3, "$name: ID=$id skipped (developId=y)";
-				push (@skippedId, $devid);
-				next;
+			# wenn es keine developId gibt, dann die folgenden Abfragen ueberspringen
+			if (defined($ProtocolListSIGNALduino{$id}{developId}))
+			{
+				if ($ProtocolListSIGNALduino{$id}{developId} eq "m") {
+					if ($develop eq "" || $develop !~ m/m$id/) {  # ist nur zur Abwaertskompatibilitaet und kann in einer der naechsten Versionen entfernt werden
+						push (@devModulId, $id);
+					}
+				}
+				elsif ($ProtocolListSIGNALduino{$id}{developId} eq "p") {
+					SIGNALduino_Log3 $name, 3, "$name IdList: ID=$id skiped (developId=p), not for user";
```

Comment:
Ich glaube dass wir über diese Formulierung schon einmal diskutiert hatten. not for user finde ich nicht gut.
Wenn ich eine Protokoll Definition neu anlege und es dann einem Anwender zum verifizieren gebe, wieso soll er es dann nicht verwenden?

Reply:
Protokolle die die developId p haben sind noch nichts für den user. Damit können auch Protokoll Ids gekennzeichnet werden, die nur zum Senden sind.
Die developId p sind nur für Entwickler oder die wo beim Entwickeln helfen wollen.

Wenn sie soweit funktionsfähig sind, das sie ein user testen kann, bekommen sie developId y

Reply:
Ein Anwender ist für mich doch jemand der das verwenden will und kein Entwickler.

Und wenn jetzt noch nicht mal von dem Anwender der das Gerät hat getestet wurde, packe ich ein p rein. Er ist aber doch immer noch ein Anwender und kein Entwickler oder?

Reply:
Dies ist dann doch kein normaler Anwender mehr. Du wirst Ihm wahrscheinlich gesagt haben, er soll zum testen auf die dev-r33 updaten und die entspechende ID in die whitelist eintragen

Reply:
so würde ich es dann machen. Anschließend wundert er sich, dass er es nicht für ihn gedacht ist.

Reply:
wieso schreiben wir nicht hin für welchen Fall es ist, anstelle davon, dass es nicht für den Anwender ist. also anstelle "not for user" "use only for testing and signal development" ?

Reply:
> Anschließend wundert er sich, dass er es nicht für ihn gedacht ist.

Die Meldung bekommt er nur, wenn er keine whitelist verwendet.